### PR TITLE
e2e File plugin test scenarios

### PR DESCRIPTION
**Pull request by sawantpritam**
All the review comments from old PR https://github.com/cdapio/hydrator-plugins/pull/1589 have been addressed in this PR.

@rmstar @itsankit-google Please review.

### CODE REVIEW

File: core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count is equal to IN record count
+    Then Validate output file generated by file sink plugin is equal to expected output file "<ExpectedOutputFile>"
+    Examples:
+      | FileFormat | ExpectedOutputFile    |
+      | csv        | outputForCSVInputTest |
+      | tsv        | tsvFormatOutput       |
+      | json       | jsonFormatOutput      |
+      | orc        | orcFormatOutput       |
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature:48]: Can we add validations for `avro` and `parquet` format as well?

**Review by itsankit-google — COMMENTED** [review]
Do we have tests for `Advanced Properties` in File Source like `Regex Path Filter` and `Path Field`??

File: core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature (hunk 1 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature:48]: Facing difficulty in comparing expected outputs for avro and parquet. 

If it is ok, for now we can add scenario for avro and parquet just to validate pipeline gets succeeded excluding validation step https://github.com/cdapio/hydrator-plugins/blob/8558303b282fa815b346e371a8026986d6147044/core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature#L42

**Comment by sawantpritam** [comment]
> Do we have tests for `Advanced Properties` in File Source like `Regex Path Filter` and `Path Field`??

Added Scenario for `Path Field`. `Regex Path Filter` scenario already exists.

File: core-plugins/src/e2e-test/features/fileplugin/source/FileToFile.feature [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+    When Select plugin: "File" from the plugins list as: "Sink"
+    Then Connect plugins: "File" and "File2" to establish connection
+    Then Navigate to the properties page of plugin: "File"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "outputFieldTestFile"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Click plugin property: "skipHeader"
+    Then Enter input plugin property: "pathField" with value: "outputPathField"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "outputFieldTestFileSchema"
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/features/fileplugin/source/FileToFile.feature]: nit : Rename `outputField` to `pathField` everywhere to make it intuitive.

File: core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+  Scenario: To verify data is getting transferred from File to File sink successfully for delimited format
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature:95]: For delimited, csv & tsv formats, do we have tests for `Enable Quoted Values`?
image

File: core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature (hunk 1 above)
Comment (itsankit-google) [reply core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature:48]: @rmstar wdyt??

File: core-plugins/src/e2e-test/resources/testdata/file/expected_outputs/OUTPUT_FOR_TSV.csv [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+TS-GCS-DSGN-01-01,Verify the path field name only field with true value to check whether to use file name only instead of URI,TC-GCS-DSGN-23,positive
+TS-GCS-DSGN-01-01,Verify the GCS properties of Read files recursively field with true or false value,TC-GCS-DSGN-24,positive
+TS-GCS-DSGN-01-01,Verify the GCS properties of Data file encrypted field with true or false value,TC-GCS-DSGN-25,positive
+TS-GCS-DSGN-01-01,Verify the GCS properties of encryption metadata file suffix field,TC-GCS-DSGN-26,positive
+TS-GCS-DSGN-01-01,Verify the GCS properties of File system properties field which is Additional properties to use with the InputFormat when reading the data,TC-GCS-DSGN-27,positive
+TS-GCS-DSGN-01-01,Verify the GCS properties of File encoding field(UTF-8),TC-GCS-DSGN-28,positive
+TS-GCS-DSGN-01-01,Verify the maximum split size for each partion in bytes in Advance GCS properties,TC-GCS-DSGN-29,positive
+TS-GCS-DSGN-01-01,Verify the minimum split size for each partion in bytes in Advance GCS properties,TC-GCS-DSGN-30,positive
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/resources/testdata/file/expected_outputs/OUTPUT_FOR_TSV.csv]: Do we need test data with so many rows??

Comment (sawantpritam) [reply core-plugins/src/e2e-test/resources/testdata/file/expected_outputs/OUTPUT_FOR_TSV.csv]: No not required. Please suggest maximum row count. 10 ?

File: core-plugins/src/e2e-test/features/fileplugin/source/FileToFile.feature (hunk 2 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/features/fileplugin/source/FileToFile.feature]: Renamed.

File: core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature (hunk 3 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature:95]: Added tests for `Enable Quoted Values`

File: core-plugins/src/e2e-test/resources/testdata/file/expected_outputs/OUTPUT_FOR_TSV.csv (hunk 4 above)
Comment (sawantpritam) [reply core-plugins/src/e2e-test/resources/testdata/file/expected_outputs/OUTPUT_FOR_TSV.csv]: Reduced rows.

File: core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature (hunk 1 above)
Comment (rmstar) [reply core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature:48]: What's the issue with comparing output for avro and parquet?

> for now we can add scenario for avro and parquet just to validate pipeline gets succeeded excluding validation step

If you're blocked on this just validating successful execution of the pipeline is ok for now. Please file a Jira for adding support for comparing avro/parquet files and add it link to it in the PR

Comment (sawantpritam) [reply core-plugins/src/e2e-test/features/fileplugin/sink/FileToFile.feature:48]: Added test scenario to validate successful execution of pipeline for avro and parquet format.
Created jira to add validation step for the comparison of expected outputs. 
https://cdap.atlassian.net/browse/CDAP-19598

File: core-plugins/src/e2e-test/java/io/cdap/plugin/file/stepsdesign/FilePlugin.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Then("Enter File plugin property: override field name with value: {string}")
+  public void enterFilePluginPropertyOverrideFieldNameWithValue(String fieldName) {
+    FileActions.enterOverrideFieldName(PluginPropertyUtils.pluginProp(fieldName));
+  }
+
+  @Then("Select File plugin property: override field data type with value: {string}")
+  public void selectFilePluginPropertyOverrideFieldDataTypeWithValue(String dataType) {
+    FileActions.selectOverrideFieldDatatype(PluginPropertyUtils.pluginProp(dataType));
+  }
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/java/io/cdap/plugin/file/stepsdesign/FilePlugin.java]: Both these steps looks similar, can they be combined to one like - 
```
"Select File plugin property: override field {string} with value: {string}"
```

Comment (sawantpritam) [reply core-plugins/src/e2e-test/java/io/cdap/plugin/file/stepsdesign/FilePlugin.java]: One step is to `Enter` field name in textbox and other is to `Select` datatype from dropdown.

File: core-plugins/src/e2e-test/java/io/cdap/plugin/file/actions/FileActions.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    SeleniumHelper.getPropertiesLocators(FileLocators.class);
+  }
+
+  public static void enterOverrideFieldName(String fieldName) {
+    ElementHelper.sendKeys(FileLocators.overrideFieldName, fieldName);
+  }
+
+  public static void selectOverrideFieldDatatype(String dataType) {
+    ElementHelper.selectDropdownOption(FileLocators.overrideFieldDatatype,
+                                       CdfPluginPropertiesLocators.locateDropdownListItem(dataType));
```
Comment (itsankit-google) [inline core-plugins/src/e2e-test/java/io/cdap/plugin/file/actions/FileActions.java]: What if we want to override data type of more than one fields??

Comment (sawantpritam) [reply core-plugins/src/e2e-test/java/io/cdap/plugin/file/actions/FileActions.java]: Added support to override datatype of more than one field in single step.

**Review by itsankit-google — APPROVED** [review]

**Pull request merged**